### PR TITLE
Fixed Issue 332, added parser, lexer rules and code stubs for "rec" declaration.

### PR DESCRIPTION
**Pull request by sychoo**

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>wyvern-forked</name>
```
Comment (JonathanAldrich) [inline .project]: This probably doesn't belong in the PR, right?

File: examples/modules/arrowModuleClient1.wyv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import modules.arrowModuleImplementation1
```
Comment (JonathanAldrich) [inline examples/modules/arrowModuleClient1.wyv]: Can you put these test files in tools/src/wyvern/tools/tests/modules?

Although the current project isn't completely consistent about this, I'd like to keep examples/ for files that do something interesting.  (Alternatively, you can work this test case into an example that does something interesting)

File: tools/src/wyvern/target/corewyvernIL/decl/RecConstructDeclaration.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import wyvern.tools.errors.FileLocation;
+
+public class RecConstructDeclaration extends DeclarationWithRHS {
+
+  private ValueType type;
+
+  public RecConstructDeclaration(String fieldName, ValueType type, IExpr iExpr, FileLocation loc) {
+    super(fieldName, iExpr, loc);
+    this.type = type;
+    System.out.println();
```
Comment (JonathanAldrich) [inline tools/src/wyvern/target/corewyvernIL/decl/RecConstructDeclaration.java]: remove printlns before this is merged

File: tools/src/wyvern/target/corewyvernIL/decltype/RecConstructDeclType.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package wyvern.target.corewyvernIL.decltype;
+
+import wyvern.target.corewyvernIL.IASTNode;
+import wyvern.target.corewyvernIL.astvisitor.ASTVisitor;
+import wyvern.target.corewyvernIL.support.FailureReason;
+import wyvern.target.corewyvernIL.support.TypeContext;
+import wyvern.target.corewyvernIL.support.View;
+import wyvern.target.corewyvernIL.type.ValueType;
+
+public class RecConstructDeclType extends DeclTypeWithResult implements IASTNode {
```
Comment (JonathanAldrich) [inline tools/src/wyvern/target/corewyvernIL/decltype/RecConstructDeclType.java]: Might be able to just use ValDeclType or equivalent) for this.

File: .project (hunk 1 above)
Comment (sychoo) [reply .project]: Hi Jonathan! This does not belong in the PR. It is a cache file generated by my IDE. I will make sure to clean it up.
